File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/adapters/LoanBookAdapter.java
package com.hgianastasio.biblioulbrav2.views.adapters;

import android.content.Context;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.hgianastasio.biblioulbrav2.R;
import com.hgianastasio.biblioulbrav2.models.loanbooks.LoanBookModel;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by heitor_12 on 04/05/17.
 */

public class LoanBookAdapter extends RecyclerView.Adapter<LoanBookAdapter.ViewHolder> {
    private List<LoanBookModel> list;
    private Context mContext;
    private OnItemClickListener onItemClickListener;

    public LoanBookAdapter(List<LoanBookModel> list, Context mContext) {
        this.list = list;
        this.mContext = mContext;
    }

    public interface OnItemClickListener{
        void onItemClick(LoanBookModel loanBookModel);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.loan_book_item,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        LoanBookModel model = list.get(position);
        holder.tvBookTitle.setText(model.getTitle());
        holder.tvDeadline.setText(String.format("Prazo: %s",model.getDeadline()));
        holder.tvBookPenalty.setText(String.format("Multa: R$%s",model.getPenalty()));
        if (onItemClickListener!=null)
            holder.root.setOnClickListener(v->onItemClickListener.onItemClick(model));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }

    class ViewHolder extends RecyclerView.ViewHolder{
        @BindView(R.id.tvBookTitle) TextView tvBookTitle;
        @BindView(R.id.tvDeadline) TextView tvDeadline;
        @BindView(R.id.tvBookPenalty) TextView tvBookPenalty;
        @BindView(R.id.rlRoot) RelativeLayout root;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }
    }
}

File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/adapters/LoanBookAdapter.kt
package com.hgianastasio.biblioulbrav2.views.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.hgianastasio.biblioulbrav2.R
import com.hgianastasio.biblioulbrav2.databinding.LoanBookItemBinding
import com.hgianastasio.biblioulbrav2.models.loanbooks.LoanBookModel

/**
 * Created by heitor_12 on 04/05/17.
 */
class LoanBookAdapter(private val list: List<LoanBookModel?>, private val mContext: Context?) : RecyclerView.Adapter<LoanBookAdapter.ViewHolder>() {
    private var onItemClickListener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(loanBookModel: LoanBookModel?)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.loan_book_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = list[position]
        holder.binding.tvBookTitle.text = model?.title
        holder.binding.tvDeadline.text = String.format("Prazo: %s", model?.deadline)
        holder.binding.tvBookPenalty.text = String.format("Multa: R$%s", model?.penalty)
        holder.binding.root.setOnClickListener {  onItemClickListener?.onItemClick(model) }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
        this.onItemClickListener = onItemClickListener
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val binding = LoanBookItemBinding.bind(itemView)
    }

}